lib/chef/http/authenticator.rb

```diff
@@ -104,6 +152,51 @@ def load_signing_key(key_file, raw_key = nil)
         raise Chef::Exceptions::InvalidPrivateKey, msg
       end
 
+      def self.retrieve_certificate_key(client_name)
+        require "openssl" unless defined?(OpenSSL)
+
+        if ChefUtils.windows?
+          # This code block assumes a certificate with a subject name like "Chef-<node-name>" is in the \LocalMachine\My store and
+          # that there is a password stored in the registry to be used to export the pfx with.
+          powershell_password_code = <<~CODE
+              Try {
```

Comment:
There's a Win32::Registry module that might help here instead of having ot shell out directly

Reply:
done